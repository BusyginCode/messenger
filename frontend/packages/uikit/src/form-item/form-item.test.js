import React from 'react'
import { render } from 'enzyme'
import FormItem from './form-item'

describe('uikit/FormItem', () => {
  it('should render correctly', () => {
    const formItem = render(<FormItem message="message" label="label" labelRight="labelRight" name="name" />, {
      context: { formik: {} }
    })
    expect(formItem).toMatchSnapshot()
  })

  it('should render with error correctly', () => {
    const formItem = render(<FormItem name="name" message="message" />, {
      context: {
        formik: {
          touched: { name: true },
          values: { name: 'name' },
          errors: { name: 'error' }
        }
      }
    })
    expect(formItem).toMatchSnapshot()
  })

  it('should render with success correctly', () => {
    const formItem = render(<FormItem name="name" successMessage="success" />, {
      context: { formik: { touched: { name: true }, values: {} } }
    })
    expect(formItem).toMatchSnapshot()
  })

  it('should render with isSuccess correctly', () => {
    const formItem = render(<FormItem name="name" isSuccess />, {
      context: { formik: { touched: { name: true }, values: {} } }
    })
    expect(formItem).toMatchSnapshot()
  })
})
