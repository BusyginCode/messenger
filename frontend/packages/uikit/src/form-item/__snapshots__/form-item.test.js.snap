// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`uikit/FormItem should render correctly 1`] = `
<div
  class="formItem m row"
  data-role="formItem_name"
>
  <p
    breed="subtitle"
    class="label label base general m left semibold l_l"
    data-role="formItem_label_label"
  >
    label
  </p>
  <div
    class="wrapper size-l"
  >
    <input
      class="input default"
      data-role="formItem_control_name"
      name="name"
      type="text"
    />
  </div>
  <p
    class="labelRight base general s left"
    data-role="formItem_label_right_label_right"
  >
    labelRight
  </p>
  <p
    class="message base general s left"
    data-role="formItem_message_message"
  >
    message
  </p>
</div>
`;

exports[`uikit/FormItem should render with error correctly 1`] = `
<div
  class="formItem m danger"
  data-role="formItem_name"
>
  <div
    class="wrapper size-l"
  >
    <input
      class="input danger"
      data-role="formItem_control_name"
      name="name"
      type="text"
      value="name"
    />
  </div>
  <p
    class="message base general danger s left"
    data-role="formItem_message_error"
  >
    error
  </p>
</div>
`;

exports[`uikit/FormItem should render with isSuccess correctly 1`] = `
<div
  class="formItem m success"
  data-role="formItem_name"
>
  <div
    class="wrapper size-l"
  >
    <input
      class="input default"
      data-role="formItem_control_name"
      name="name"
      type="text"
    />
  </div>
</div>
`;

exports[`uikit/FormItem should render with success correctly 1`] = `
<div
  class="formItem m success"
  data-role="formItem_name"
>
  <div
    class="wrapper size-l"
  >
    <input
      class="input success"
      data-role="formItem_control_name"
      name="name"
      type="text"
    />
  </div>
  <p
    class="message base general success s left"
    data-role="formItem_message_success"
  >
    success
  </p>
</div>
`;
